src/addresses.py

```
@@ -0,0 +1,103 @@
+    applicable:
+        * `postal_code_city`: the city inferred from the zip code
+        * `postal_code_state`: the state inferred from the zip code
+    """
+    (postal_code_city, postal_code_state
+        ) = get_city_and_state_from_postal_code(country, postal_code)
+    return Address(
+        address = street_address,
+        city = city,
+        state = state,
+        postal_code = postal_code,
+        postal_code_city = postal_code_city,
+        postal_code_state = postal_code_state,
+        country=country.lower()
+    )
+
+def get_api_headers(api_name: str) -> Dict:
+    if api_name == 'ziptastic':
+        return { "x-key" : config.ZIPTASTIC_API_KEY }
+    if api_name == 'postcodes':
+        return {}
+    return {}
+
+def get_city_and_state_from_postal_code(
+    country_name: str, # the two letter code for the country
+    postal_code: str
+) -> Tuple[Optional[str], Optional[str]]:
+    """ Given a country code and a zip, fetch the city and state."""
+    standardized_country_code = country_name.upper()
+    if standardized_country_code == 'GB' or standardized_country_code == 'UK':
+        resp = requests.get(
+            BASE_POSTCODES_API_URL+postal_code,
+            headers = get_api_headers('postcodes')
+        )
+        rjson = resp.json()
+        result = rjson.get('result', {})
+        city = result['admin_district'] if 'admin_district' in result else None
+        return (city, None)
+    elif (
+        standardized_country_code == 'US'
+        or standardized_country_code == 'CA'
+        or standardized_country_code == 'fr'
+    ):
+        # FYI: Other countries Ziptastic supports:
+        # https://www.getziptastic.com/faq
+        try:
+            resp = requests.get(
+                f'{BASE_ZIPTASTIC_API_URL}{standardized_country_code}/{postal_code}',
+                headers = get_api_headers('ziptastic')
+            )
+            rjson = resp.json()
+        except:
```

this line will catch any exception that occurs in the try block, which might mask unexpected errors. can we be more specific? what's the exception you're expecting to catch in this case? you can see some of the exceptions the requests library throws [here](https://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module)